src/UserPreferencesDB.py

```diff
@@ -0,0 +1,129 @@
+""" Class to Create a User Preferences Databases """
+import sqlite3
+import logging
+
+logger = logging.getLogger(__name__) # Starting Logger
+
+class UserPreferencesDB:
+    """ class to create a user preferences database """
+    # As this will likely make it easier to test during testing
+    # Return Codes:
+    #   1 indicates ran correctly
+    #   2 indicates operational Error
+
+    def __init__(self, database_name: str):
+        """ Initializing the database connection/creating database """
+        logger.info("Initializing the database connection/creating database")
+        try:
+            self.database = sqlite3.connect(database_name)
+            self.create_table()
+            logger.info("Database connection successful")
+        except sqlite3.OperationalError:
+            logger.error("Did not create Database object")
+
+    def create_table(self) -> int:
+        """ Creating table in database"""
+        logger.info("Creating table in database")
+        try:
+            cursor = self.database.cursor()
+            cursor.execute(
+                "CREATE TABLE IF NOT EXISTS user_preferences ("
+                    "username TEXT UNIQUE PRIMARY KEY, "
+                    "default_dir TEXT"
+                ")"
+            )
+            self.database.commit()
+            logger.info("Table created successfully")
+            return 1
+        except sqlite3.OperationalError:
+            logger.error("Table did not get created")
+            return 2
```

Comment:
by catching and not raising the exception, the calling function will only have an exception if the connect raises an exception.  And it isn't looking at the return value.  If you are prepared to catch exceptions at a higher level and you aren't going to process the return value, there's no reason to return a value, just re-raise an exception or don't catch them and allow them to be processed at a higher level.